cpp/open3d/core/SizeVector.h

```diff
@@ -83,7 +144,40 @@ class SizeVector : public std::vector<int64_t> {
                 });
     }
 
+    int64_t GetLength() const {
+        if (size() == 0) {
+            utility::LogError("Cannot get length of a 0-dimensional shape.");
+        } else {
+            return operator[](0);
+        }
+    }
+
     std::string ToString() const { return fmt::format("{}", *this); }
+
+    void AssertCompatible(const DynamicSizeVector& dsv,
+                          const std::string msg = "") const {
+        if (!IsCompatible(dsv)) {
+            if (msg.empty()) {
+                utility::LogError("Shape {} is not compatible with {}.",
+                                  ToString(), dsv.ToString());
+            } else {
+                utility::LogError("Shape {} is not compatible with {}: {}",
+                                  ToString(), dsv.ToString(), msg);
+            }
+        }
+    }
+
```

Comment:
It might cause confusion if we use `==` since `None` can match any value.

E.g. it is intuitive that `(3, 4) == (3, 4)`; but it is less intuitive that `(3, 4) == (None, None)`